core/peer/peer_test.go

```diff
@@ -298,6 +300,47 @@ func TestDeliverSupportManager(t *testing.T) {
 	require.NotNil(t, chainSupport, "chain support should not be nil")
 }
 
+func TestConfigCallback(t *testing.T) {
+	peerInstance, cleanup := NewTestPeer(t)
+	defer cleanup()
+
+	var callbackInvoked bool
+	peerInstance.AddConfigCallbacks(func(bundle *channelconfig.Bundle) {
+		callbackInvoked = true
+		orderer, exists := bundle.OrdererConfig()
+		require.True(t, exists)
+		require.NotEmpty(t, orderer.Organizations()["SampleOrg"].Endpoints)
+	})
+
+	peerInstance.Initialize(
+		nil,
+		nil,
+		plugin.MapBasedMapper(map[string]validation.PluginFactory{}),
+		&ledgermocks.DeployedChaincodeInfoProvider{},
+		nil,
+		nil,
+		runtime.NumCPU(),
+	)
+
+	testChannelID := fmt.Sprintf("mytestchannelid-%d", rand.Int())
+	block, err := configtxtest.MakeGenesisBlock(testChannelID)
+	if err != nil {
+		fmt.Printf("Failed to create a config block, err %s\n", err)
+		t.FailNow()
+	}
+
+	// We expect the callback to be invoked when the channel is created
+	require.False(t, callbackInvoked)
+
+	err = peerInstance.CreateChannel(testChannelID, block, &ledgermocks.DeployedChaincodeInfoProvider{}, nil, nil)
```

Comment:
why use t.Fatalf if you have require somewhere else? Let's be consistent with the pattern

Reply:
Good point.  Careless cut and paste from a previous test.  I've changed to just using require.